### ISSUE TITLE
Syntax highlighting problem, unreachable code or unbalanced brackets

### ISSUE THREAD

**Issue by d-a-s** [Microsoft/vscode#64393]
Issue Type: <b>Bug</b>

I was working on a section of code, and when I went to the top to add a helper function, I found that as soon as I typed `return` in the new function's body, before I could type the rest of the return statement and the closing bracket for the function, much of the code disappeared (see screen shot). 
![screenshot_2018-12-04_19-03-01](https://user-images.githubusercontent.com/26210424/49481130-76fbdb00-f7f7-11e8-81ef-bbf678f8de53.png)

As soon as I added the closing bracket the code appeared correctly again. The code seemed to disappear/reappear around the same time as the code highlighting changed.

I tried to get a smaller block of code to reproduce, but this was as small as I could get it
```
(function () {
  function no(i) { return 
  function nu(i) { return i !== undefined };
  function Map() {
    this._kv = [];
    return this;
  };
  var mprot = Map.prototype;
  mprot.size = function () { return this._kv.length };
  mprot._findi = function (key) {
    for (var i = 0; i < this._kv.length; i++) {
      if (this._kv[i].key === key) return i;
    }
  };
  mprot.get = function (key) {
    var i = this._findi(key);
    if (nu(i)) return this._kv[i];
  };
  mprot.has = function (key) {
    return nu(this._findi(key));
  };
  mprot.set = function (key, value) {
    var obj = this.get(key);
    if (!obj) {
      obj = { key: key, value: undefined };
      this._kv.push(obj);
    }
    obj.value = value;
    return this;
  };
  mprot.delete = function (key) {
    var i = this._findi(key);
    if (nu(i)) {
      this._kv.splice(i, 1);
      return true;
    }
    return false;
  };
  if (!('Map' in window)) window.Map = Map;
  else window.Map2 = Map;
})();
```

-----

VS Code version: Code 1.29.1 (bc24f98b5f70467bc689abf41cc5550ca637088e, 2018-11-15T19:07:43.495Z)
OS version: Linux x64 4.18.11-041811-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 2837)|
|GPU Status|2d_canvas: unavailable_software<br>checker_imaging: disabled_off<br>flash_3d: unavailable_software<br>flash_stage3d: unavailable_software<br>flash_stage3d_baseline: unavailable_software<br>gpu_compositing: unavailable_software<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: unavailable_software<br>video_decode: unavailable_software<br>video_encode: unavailable_software<br>webgl: enabled_readback<br>webgl2: unavailable_off|
|Load (avg)|1, 1, 1|
|Memory (System)|15.15GB (5.28GB free)|
|Process Argv|--unity-launch --disable-gpu|
|Screen Reader|no|
|VM|0%|

</details>Extensions: none

**Comment by mjbvz** [comment]
Please test this in the first VS Code 1.31 insiders build once it comes out at the end of this week. This looks like a potential electron issue and 1.31 includes a new version of electron

**Comment by bpasero** [comment]
Changing the labels, `electron-3.0.x-update` should be used to signal new regressions we got by updating to Electron 3 :+1:

**Comment by d-a-s** [comment]
Yes, the problem appears to be fixed in VS Code 1.31 insiders build. Thanks!

**Comment by bpasero** [comment]
Leaving open until we ship Electron 3.0.x to stable.
